swift/StableDiffusion/pipeline/Encoder.swift

```diff
@@ -49,83 +38,71 @@ public struct Encoder: ResourceManaging {
     /// Prediction queue
     let queue = DispatchQueue(label: "encoder.predict")
 
-    /// Batch encode latent samples into images
-    /// - Parameters:
-    ///   - image: image used for image2image
-    ///   - diagonalNoise: random noise for `DiagonalGaussianDistribution` operation
-    ///   - noise: random noise for initial latent space based on strength argument
-    ///   - alphasCumprodStep: calculations using the scheduler traditionally calculated in the pipeline in pyTorch Diffusers library.
-    /// - Returns: The encoded latent space as MLShapedArray
+    /// Encode image into latent sample
+    ///
+    ///  - Parameters:
+    ///    - image: Input image
+    ///    - scaleFactor
+    ///    - random
+    ///  - Returns: The encoded latent space as MLShapedArray
     public func encode(
-        image:  CGImage,
-        diagonalNoise: MLShapedArray<Float32>,
-        noise: MLShapedArray<Float32>,
-        alphasCumprodStep: AlphasCumprodCalculation
+        _ image: CGImage,
+        scaleFactor: Float32 = 0.18215,
```

Comment:
@GuiyeC  Could you please pull this into the `StableDiffusionPipeline.Configuration.swift` and also couple it with the corresponding scale factor in `Decoder.swift`? The decoder scale factor is also hardcoded to 0.18215 and we should make it configurable. Thanks!

Reply:
@atiorh I moved those into the Configuration struct, I had them be two different scale factors as I have seen that they can be different in some cases, for example when using Instruct Pix2Pix the encoder needs a scale factor of 1 but the decoder uses the same 0.18215 that was hard coded.
Let me know if you want some more changes, thank you for the review!